torchdata/dataloader2/README.md

```diff
@@ -0,0 +1,68 @@
+# DataLoader2 (Prototype)
+
+`DataLoader2` is introduced to execute `DataPipe` graph, with support for
+
+- dynamic sharding for multi-process and distributed data loading
+- multiple backend `ReadingServices` (WIP for `DistributedReadingService`)
+- `DataPipe` graph in-place modification like shuffle control, memory pinning, etc.
+- snapshot the state of data-preprocessing pipeline (WIP)
+
+## Why DataLoader2
+
+`DataPipe` is introduced to not only decompose the data pre-processing operations, but also decouple the overloaded
+data-manipulation features from `DataLoader`. So, a light-weight `DataLoader2` is required. Besides, a certain features
+can only be achieved with `DataLoader2` like snapshotting and bridging data-processing pipeline between research and
+production.
+
+These options are configured by the constructor arguments of a `DataLoader2`, which has signature:
+
+```py
+DataLoader2(datapipe, datapipe_adapter_fn=None, reading_service=None)
+```
+
+The sections below describe in details the effects and usages of these options.
+
+## DataPipe
+
+`DataLoader2` supports both `MapDataPipe` and `IterDataPipe`. To understand the basic structure of `DataPipe`, please
+see [What are DataPipes?](https://github.com/pytorch/data#what-are-datapipes)
+
```

Comment:
It may be worth mentioning that it doesn't support `Dataset` and `IterableDataset`? Unless they put a wrapper around it?

Reply:
Sounds good.